internal/tool/generate/generator.go

```diff
@@ -422,6 +423,27 @@ func extractComponent(opt Options, pkg *packages.Package, file *ast.File, tset *
 					formatType(pkg, arg))
 			}
 			refs = append(refs, named)
+		} else if isWeaverListener(t) {
+			var lisName string
+			if f.Tag != nil {
+				tag := f.Tag.Value
+				const tagPrefix = "`weaver:\""
+				const tagSuffix = "\"`"
+				if !strings.HasPrefix(tag, tagPrefix) || !strings.HasSuffix(tag, tagSuffix) {
+					return nil, errorf(pkg.Fset, f.Pos(),
+						"Listener tag must be of the format `weaver:\"name\"`, got %s",
+						tag)
+				}
+				lisName = strings.TrimSuffix(strings.TrimPrefix(tag, tagPrefix), tagSuffix)
```

Comment:
Do we need to check that these tag names look like what we expect. For example, a user might do something like 


```go
type foo struct {
    weaver.Implements[Foo]
    x weaver.Listener `weaver:" 1 2 3 "`
}
```

Reply:
Ack, will do in a separate PR.